ipaserver/plugins/cert.py

```diff
@@ -771,6 +771,17 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
         cn = cns[-1].value  # "most specific" is end of list
 
         if principal_type in (SERVICE, HOST):
+
+            has_dns_in_san_ext = False
+            if ext_san:
+                for gn in x509.process_othernames(ext_san.value):
+                    if isinstance(gn, cryptography.x509.general_name.DNSName):
+                        has_dns_in_san_ext = True
+
+            if not ext_san or not has_dns_in_san_ext:
+                print('Warning: The SAN extension '
```

Comment:
Also recalling the review comments from https://github.com/freeipa/freeipa/pull/773#pullrequestreview-37222207 about sending the warning back to the client, and the `print` is just placeholder while final version
of patch is worked out.